kernel/include/timeout_q.h

```diff
@@ -37,6 +37,14 @@ static inline bool z_is_inactive_timeout(const struct _timeout *to)
 	return !sys_dnode_is_linked(&to->node);
 }
 
+static inline bool z_is_expired_timeout(const struct _timeout *to)
+{
+	/* When timeout is aborted then dticks is not reset. It can be used to detect
+	 * if timeout was aborted or expired.
+	 */
```

Comment:
Although it is true that when a timeout is aborted _dticks_ is not reset, the value of _dticks_ is insufficient to determine if the timeout was aborted or expired.

Consider the following scenario ... _sys_clock_announce()_ is called and there are multiple timeouts ready to expire. That is, there are multiple timeouts whose _dticks_ field is set to 0. The handlers for each of these timeouts are called as each timeout is processed. Should one of the earlier timeouts abort one of the later timeouts that are ready to expire, then that later timeout will get aborted while its _dticks_ is 0 and its handler will not execute.